tutorials/persistance-comparison-series/BinaryWriterReader/binary_reader_writer.md

```diff
@@ -0,0 +1,426 @@
+
+Finally, as before, we create a `FileStream` (5) and `BinaryWriter` (6) and use `Write()` (7) to write the `jsonString` into the file.
+
+Then, when the game starts (8), we need to read the data back into the hit count fields:
+
+```cs
+private void Start() // 8
+{
+    // Check if the file exists to avoid errors when opening a non-existing file.
+    if (File.Exists(HitCountFile)) // 9
+    {
+        // Open a stream to the file that the `BinaryReader` can use to read data.
+        // They need to be disposed at the end, so `using` is good practice
+        // because it does this automatically.
+        using FileStream fileStream = File.Open(HitCountFile, FileMode.Open); // 10
+        using BinaryReader binaryReader = new(fileStream); // 11
+
+        // 12
+        var jsonString = binaryReader.ReadString();
+        var hitCount = JsonUtility.FromJson<HitCount>(jsonString);
+
+        // 13
+        if (hitCount != null)
+        {
+            // 14
+            hitCountUnmodified = hitCount.Unmodified;
+            hitCountShift = hitCount.Shift;
+            hitCountControl = hitCount.Control;
+        }
+    }
+}
+```
+
+We check if the file exists first (9). In case it does, we saved data before and can proceed reading it.
+
+Using a `FileStream` again (10) with `FileMode.Open`, we create a `BinaryReader` (11). Since we are reading a json string, we need to use `ReadString()` (12) this time and then transform it via `FromJson<HitCount>()` into a `HitCount` object.
+
+If this worked out (13), we can then extract `hitCountUnmodified`, `hitCountShift`, and `hitCountControl` from it (14).
+
+Note that the data is saved in a binary format, which is, of course, not safe. Tools to read binary are available and easy to find. For example, this `BinaryReaderWriterExampleJson` file read with `bless` would result in this:
+
+<img src="images/09_json_decoded.png" alt="File BinaryReaderWriterExampleJson" width=""/>
+
+You can clearly identify the three values we saved. While the `BinaryReader` and `BinaryWriter` are a simple and easy way to save data and they at least offer a way so that the data is not immidiately readable, they are by no means safe.
+
+In a future tutorial, we will look at encryption and how to improve safety of your data along with other useful features like migrations and performance improvements.
+
+## Conclusion
+
+In this tutorial, we learned how to utilize `BinaryReader` and `BinaryWriter` to save data. `JsonUtility` helps structure this data. They are simple and easy to use, and not much code is required.
+
+What are the downsides, though?
+
+First of all, we open, write to, and save the file every single time the capsule is clicked. While not a problem in this case and certainly applicable for some games, this will not perform very well when many save operations are made when your game gets a bit more complex.
+
+Also, the data is saved in a readable format and can easily be edited by the player.
```

Comment:
Probably this is less true here than it is than with the File approach, but hex readers are quite common 😁